cmd/sonrd/didutil/new_did.go

```
@@ -0,0 +1,112 @@
+package didutil
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+
+	"github.com/AlecAivazis/survey/v2"
+	"github.com/cosmos/cosmos-sdk/crypto/keys/ed25519"
+	"github.com/cosmos/cosmos-sdk/crypto/keys/secp256k1"
+	"github.com/sonr-io/sonr/pkg/did"
+	"github.com/sonr-io/sonr/pkg/did/ssi"
+)
+
+func surveyNewDid() error {
+	// Webauthn confirmation
+	webauthn := false
+	prompt := &survey.Confirm{
+		Message: "Would you like to generate a Webauthn Credential in your browser?",
+	}
+	survey.AskOne(prompt, &webauthn)
+
+	if webauthn {
+		// Open browser and serve index.html
+	} else {
+		pubKey := ""
+		// Present public key types
+		selectOpt := &survey.Select{
+			Message: "What type of key would you like to generate",
+			Options: []string{
+				"Ed25519VerificationKey2018",
+				"EcdsaSecp256k1VerificationKey2019",
+				"RsaVerificationKey2018",
+			},
+			Default: "Ed25519VerificationKey2018",
+		}
+		// perform the questions
+		survey.AskOne(selectOpt, &pubKey, nil)
+
+		// Parse Account Addr
+		addr := genAccAddressDid()
+		didRoot, err := did.ParseDID(addr)
+		if err != nil {
+			return err
+		}
+
+		var vm *did.VerificationMethod
+
+		// Parse DID Controller
+		didController, err := did.ParseDID(fmt.Sprintf("%s#test", didRoot))
+		if err != nil {
+			return err
+		}
+
+		// Switch on pubKey
+		switch pubKey {
+		case "EcdsaSecp256k1VerificationKey2019":
+			// Generate EcdsaSecp256k1VerificationKey2019
+			priv := secp256k1.GenPrivKey()
+			pub := priv.PubKey()
+			vm, _ = did.NewVerificationMethod(*didRoot, ssi.ECDSASECP256K1VerificationKey2019, *didController, pub)
```

Don't ignore error (also apply to other cases)